extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/common/DruidKubernetesPeonClient.java

```diff
@@ -111,7 +111,8 @@ public JobResponse waitForJobCompletion(K8sTaskId taskId, long howLong, TimeUnit
                           unit
                       );
       if (job == null) {
-        return new JobResponse(job, PeonPhase.FAILED);
+        log.info("K8s job for task was not found %s", taskId);
+        return new JobResponse(null, PeonPhase.FAILED);
```

Comment:
Though this is an improvement over the current NPE, if we do this then a status won't ever be recorded for the task and it may be attempted to be rerun.  We should report the status using the taskid passed into this function.

Reply:
it still gets reported by https://github.com/apache/druid/blob/master/extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesTaskRunner.java#L184 since its a PeonPhase.FAILED status